workflow/deploy.md

```diff
@@ -0,0 +1,147 @@
+
+ * Create a [New Item](https://jenkins.canonical.com/webteam/view/all/newJob):
+   * Item name: *deploy-to-<PRODUCTION_DNS>*
+   * Copy from (at the bottom the page): *deploy-to-design.ubuntu.com*
+ * Change those parameters on the configuration page:
+   * *Build*: Update the script with the rights values
+ * Save
+
+### Jobs examples
+
+Those examples are from the design.ubuntu.com project. Please adapt the sripts before using them:
+
+#### Staging
+
+```bash
+#!/bin/bash --login
+
+# Bash strict mode
+set -euo pipefail
+
+# Settings
+export image_name="prod-comms.docker-registry.canonical.com/design.ubuntu.com:${GIT_COMMIT}"
+
+# Build docker image
+if ${clean}; then ./run clean; fi
+./run build
+docker build --pull --build-arg HTTPS_PROXY=${HTTPS_PROXY} --build-arg COMMIT_ID=${GIT_COMMIT} --tag ${image_name} .
+
+# Push to registry
+docker push ${image_name}
+
+# Deploy the new image
+kubectl set image deployment/design-ubuntu-com --namespace staging design-ubuntu-com=${image_name}
+
+# Wait for the deployment to finish
+kubectl rollout status deployment/design-ubuntu-com --namespace staging --watch
+
+echo ""
+echo "==="
+echo "Image: ${image_name}"
+echo "Deployed to https://design.staging.ubuntu.com"
+echo ""
+echo "To deploy to production, run:"
+echo "https://jenkins.canonical.com/webteam/securityRealm/commenceLogin?from=%2Fwebteam%2Fjob%2Fdeploy-to-design.ubuntu.com%2Fparambuild%2F%3Fimage_tag%3D${GIT_COMMIT}"
+echo "==="
+```
+
+#### Production
+
+```bash
+#!/bin/bash --login
+
+# Bash strict mode
+set -euo pipefail
+
+# Settings
+export base_image_name="prod-comms.docker-registry.canonical.com/design.ubuntu.com"
+
+# Release the image
+kubectl set image deployment/design-ubuntu-com --namespace production design-ubuntu-com=${base_image_name}:${image_tag}
+
+# Wait for the deployment to finish
+kubectl rollout status deployment/design-ubuntu-com --namespace production --watch
+
+echo ""
+echo "==="
+echo "Image: ${base_image_name}:${image_tag}"
+echo "Deployed to https://design.ubuntu.com"
+echo "==="
+echo ""
+
+echo "Updating ${base_image_name}:latest:"
+docker pull ${base_image_name}:${image_tag}
+docker tag ${base_image_name}:${image_tag} ${base_image_name}:latest
+docker push ${base_image_name}:latest
+
+echo ""
+echo "==="
+echo "Image: ${base_image_name}:${image_tag}"
+echo "Deployed to https://design.ubuntu.com"
+echo "And pushed to ${base_image_name}:latest"
+echo "==="
+echo ""
+```
+
+## First deployment
+
+For the first deployement go on the apply and update the line you added previously. You need to change the variable `TAG_TO_DEPLOY` with the hash of the last commit in the project. 
+
+Once this tag is updated you can run the job apply, then the staging and if nothing fails the production.
+
+If the DNS is not ready yet add a line to your `/etc/hosts` with the ip of the Kubernetes instance and your DNS.
+
+## Request new certificates
+
+[Link](https://rt.admin.canonical.com/Ticket/Create.html?Queue=5)
```

Comment:
Could we make this ever-so-slightly move verbose? Something like:

> [Submit an RT](https://rt.admin.canonical.com/Ticket/Create.html?Queue=5) to ask IS to create two new Kubernetes secrets:
>
> - "{STAGING_DOMAIN}-tls": A kubernetes secret containing the TLS key, certificate and chainfile for the staging domain (e.g. example.staging.ubuntu.com)
> - "{PRODUCTION_DOMAIN}-tls": A kubernetes secret containing the TLS key, certificate and chainfile for the production domain (e.g. example.ubuntu.com)

Ideally, links should never say "link" or "click here", they should instead contain text explaining what the link is for.